modules/xml/src/smithy4s/xml/internals/XmlByteStreamDecoderCompilerImpl.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.xml
+package internals
+
+import smithy4s.schema.Schema
+import smithy4s.schema.CachedSchemaCompiler
+import smithy4s.codecs.Reader
+import fs2._
+import fs2.data.xml._
+import fs2.data.xml.dom._
+import cats.effect.Concurrent
+import cats.syntax.all._
+import smithy4s.fs2._
+
+private[smithy4s] final case class XmlByteStreamDecoderCompilerImpl[
+    F[_]: Concurrent
+]() extends CachedSchemaCompiler.Impl[XmlByteStreamDecoder[F, *]]
+    with XmlByteStreamDecoderCompiler[F] {
+
+  type Aux[A] = XmlByteStreamDecoder[F, A]
+
+  def fromSchema[A](
+      schema: Schema[A],
+      cache: Cache
+  ): XmlByteStreamDecoder[F, A] =
+    new Reader[F, Stream[F, Byte], A] {
+      val fa = XmlDocument.Decoder.fromSchema(schema)
```

Comment:
@Baccata wanted to double check with you.. do I somehow need to pass a cache down here? I'm not exactly sure.